src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -476,8 +490,10 @@ def get_all_namespaces():
     if is_multi_npu():
         for npu in range(num_npus):
             namespace = "{}{}".format(NPU_NAME_PREFIX, npu)
-            config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
-            config_db.connect()
+            # TODO: enforce caller to provide config_db explicitly and remove its default value
+            if not config_db:
+                config_db = ConfigDBConnector()
```

Comment:
namespace argument is missing in `ConfigDBConnector` call

Reply:
Thanks! Resolved.